system-parachains/encointer/src/lib.rs

```diff
@@ -617,16 +570,82 @@ impl pallet_membership::Config<pallet_membership::Instance1> for Runtime {
 	type WeightInfo = weights::pallet_membership::WeightInfo<Runtime>;
 }
 
+/// A `HandleCredit` implementation that naively transfers the fees to the block author.
+/// Will drop and burn the assets in case the transfer fails.
+pub struct AssetsToBlockAuthor<R>(PhantomData<R>);
+
+impl<R> HandleCredit<AccountIdOf<R>, pallet_encointer_balances::Pallet<R>>
+	for AssetsToBlockAuthor<R>
+where
+	R: pallet_authorship::Config + pallet_encointer_balances::Config,
+	AccountIdOf<R>: From<polkadot_primitives::AccountId> + Into<polkadot_primitives::AccountId>,
+{
+	fn handle_credit(credit: Credit<AccountIdOf<R>, pallet_encointer_balances::Pallet<R>>) {
+		if let Some(author) = pallet_authorship::Pallet::<R>::author() {
+			// We will burn 50% of community currency fees and send 50% to the block author.
+			// reasoning: If you send 100% to the author, then the author can attempt to increase
+			// the fee rate by making transactions up to the block limit at zero cost
+			// (since they pocket the fees).
+			// In the future, fees might be collected in community treasuries instead of being
+			// burned (https://forum.polkadot.network/t/towards-encointer-self-sustainability/4195)
+			let half_amount = credit.peek() / 2;
+			let (author_credit, _burnable) = credit.split(half_amount);
```

Comment:
I don't see how this treats community token fees different from KSM. If you want to burn KSM, it needs to go into some dedicated "burn account" and periodically sent back to the Relay Chain to update the Checking account.

Reply:
I'll run a test locally to be sure....

Reply:
> I don't think Encointer needs or should have the authority to burn KSM

It is a trusted teleporter though, so it does have the authority. So, it needs to be accounted for.

Reply:
I tested the fee handling and these are my observations:
* native fees are handled correctly: they are smeared over several blocks, nothing gets burned. total issuance is constant
* CC fees are credited to the current author to 100%, so my halving logic has no effect

I guess the problem is the default OnDrop behavior which will credit the author with the remainder ???

Reply:
I added commit 6a75792911a39da420e7fd69a75aeec245845067 which demonstrates what I assumed:

* native fees are untouched by this HandleCredit implementation
* CC fees will now be split 50/50 between the current block author and a "dead" account with pubkey `0x00`

Not sure I want to leave it like that, but this way I could verify that it now works as intended

Reply:
I suggest to leave the "dead" account approach for now and revisit this when we have per-community treasuries which could collect fees depending on their asset id / CommunityIdentifier

Reply:
> Yeah I would expect `HandleImbalanceDrop` to decrease the issuance. @liamaharon could you take a look?

@joepetrowski yes, the credit should decrease the issuance when it is dropped 
- https://github.com/paritytech/polkadot-sdk/blob/6b0bc052e93b742e0813f79bc331bfa0375bb304/substrate/frame/balances/src/impl_fungible.rs#L347
- https://github.com/paritytech/polkadot-sdk/blob/4c09a0631334c3e021a30aa49a503815aa047b29/substrate/frame/support/src/traits/tokens/fungibles/imbalance.rs#L66

@brenzi can you confirm if there is an issue here with dropped amounts not changing the TI and if so provide some instructions to reproduce? Thanks.

Reply:
@liamaharon I assembled steps to reproduce and verify the current 50/50 fee split:
https://gist.github.com/brenzi/ca9e0c326198c73ff2dd0a2e74468053

I consider this done, as the question if the drop gets burned or not is no longer relevant. If there is a problem it may also be rooted in our custom fee handling elsewhere, so I'm not sure it's worth digging deeper. However, If you 'd like to make sure, just remove this line: 
https://github.com/encointer/runtimes/blob/1a655369951601c0dfb3b107c0aba721fb935081/system-parachains/encointer/src/lib.rs#L603-L604

then you could verify if 50% gets burned or not

Reply:
> the question if the drop gets burned or not is no longer relevant for the current implementation

How is this not relevant? We need to track issuance of KSM.

Reply:
Tracking this, will investigate when I get time https://github.com/paritytech/polkadot-sdk/issues/4397